pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -632,5 +636,128 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
+            sample = -1;
+        } else {
+            collapse(wires[0], sample);
+            if (reset && sample == 1) {
+                applyOperation("PauliX", wires);
+            }
+        }
+        return sample;
+    }
+
+    /**
+     * @brief Set the seed of the internal random generator
+     *
+     * @param seed Seed
+     */
+    void setSeed(const size_t seed) { rng.seed(seed); }
+
+    /**
+     * @brief Sample 0 or 1 for given probabilities
+     *
+     * @param probs Probabilities of 0.
+     */
+    auto randomSample(const PrecisionT prob_0) {
+        std::discrete_distribution<int> distrib({prob_0, 1. - prob_0});
+        return distrib(rng);
+    }
+
+    /**
+     * @brief Calculate probabilities of measuring 0 or 1 on a specific wire
+     *
+     * @param wire Wire
+     * @return std::vector<PrecisionT> Both probabilities of getting 0 or 1
+     */
+    auto probs(const std::size_t wire) -> std::vector<PrecisionT> {
+        // TODO: use Measurements.probs() instead (resolve circular dependency)
+        auto *arr = this->getData();
+        const size_t stride = pow(2, this->num_qubits_ - (1 + wire));
+        const size_t vec_size = pow(2, this->num_qubits_);
+        const auto section_size = vec_size / stride;
+        const auto half_section_size = section_size / 2;
+
+        PrecisionT prob_0{0.};
+        // zero half the entries
+        // the "half" entries depend on the stride
+        // *_*_*_*_ for stride 1
+        // **__**__ for stride 2
+        // ****____ for stride 4
+        for (size_t idx = 0; idx < half_section_size; idx++) {
+            const size_t offset = stride * 2 * idx;
+            for (size_t ids = 0; ids < stride; ids++) {
+                prob_0 += std::norm(arr[offset + ids]);
+            }
+        }
+
+        std::vector<PrecisionT> probs{prob_0, PrecisionT(1.) - prob_0};
+        return probs;
+    }
+
+    /**
+     * @brief Collapse the state vector after having measured one of the
+     * qubits
+     *
+     * @param wire Wire to collapse.
+     * @param branch Branch 0 or 1.
+     */
+    void collapse(const std::size_t wire, const bool branch) {
+        auto *arr = this->getData();
+        const size_t stride = pow(2, this->num_qubits_ - (1 + wire));
+        const size_t vec_size = pow(2, this->num_qubits_);
+        const auto section_size = vec_size / stride;
+        const auto half_section_size = section_size / 2;
+
+        // zero half the entries
+        // the "half" entries depend on the stride
+        // *_*_*_*_ for stride 1
+        // **__**__ for stride 2
+        // ****____ for stride 4
+        const size_t k = branch ? 0 : 1;
+        for (size_t idx = 0; idx < half_section_size; idx++) {
+            const size_t offset = stride * (k + 2 * idx);
+            for (size_t ids = 0; ids < stride; ids++) {
+                arr[offset + ids] = {0., 0.};
+            }
+        }
+
+        normalize();
+    }
```

Comment:
We can keep it here since this is strongly state vector oriented. But please expand the docstring to provide more context for the branch variable.